runtime/caml/domain_state.h

```diff
@@ -79,4 +79,10 @@ CAML_STATIC_ASSERT(
   #define SET_Caml_state(x) (Caml_state = (x))
 #endif
 
+#ifdef CAML_NAME_SPACE
+#define Caml_state_field(field) Caml_state->field
+#else
+#define Caml_state_field(field) Caml_state->_##field
```

Comment:
It seems wrong to include this definition. In stock OCaml, we explicitly added `_` prefixed fields to avoid conflicts with compatibility macros: https://github.com/ocaml-multicore/ocaml-multicore/pull/354/files#diff-6ef4e805c0646e6c9187ddb34e0ad3adR85. In Multicore, these compatibility macros have been deleted. Hence, multicore doesn't have the `_` prefixed field names: https://github.com/ocaml-multicore/ocaml-multicore/blob/parallel_minor_gc/runtime/caml/domain_state.h#L31-L41

So you should only need:

```
#define Caml_state_field(field) Caml_state->field
```